builtin/nomad/jobspec/releaser.go

```diff
@@ -0,0 +1,598 @@
+	}
+
+	var jobResource *sdk.StatusReport_Resource
+	for _, r := range resources {
+		if r.Type == rmResourcePromotedJobName {
+			jobResource = r
+			break
+		}
+	}
+	if jobResource == nil {
+		return nil, status.Errorf(codes.Internal, "no job resource found - cannot determine overall health")
+	}
+
+	// Create our status report
+	result := sdk.StatusReport{
+		External:      true,
+		GeneratedTime: timestamppb.Now(),
+		Resources:     resources,
+		Health:        jobResource.Health,
+		HealthMessage: jobResource.HealthMessage,
+	}
+
+	// update output based on main health state
+	step.Update("Finished building report for Nomad platform")
+	step.Done()
+
+	// NOTE(briancain): Replace ui.Status with StepGroups once this bug
+	// has been fixed: https://github.com/hashicorp/waypoint/issues/1536
+	st := ui.Status()
+	defer st.Close()
+
+	// More UI detail for non-ready resources
+	for _, resource := range result.Resources {
+		if resource.Health != sdk.StatusReport_READY {
+			st.Step(terminal.StatusWarn, fmt.Sprintf("Resource %q is reporting %q", resource.Name, resource.Health.String()))
+		}
+	}
+
+	return &result, nil
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	// List of task group names which are to be promoted
+	Groups []string `hcl:"groups,optional"`
+
+	// If true, fails the canary deployment
+	FailDeployment bool `hcl:"fail_deployment,optional"`
+}
+
+type nomadClient struct {
+	NomadClient *api.Client
+}
+
+func (r *Releaser) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(docs.FromConfig(&ReleaserConfig{}))
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description(`
+Promotes a Nomad canary deployment initiated by a Nomad jobspec deployment.
+
+If your Nomad deployment is configured to use canaries, this releaser plugin lets
+you promote (or fail) the canary deployment. You may also target specific task
+groups within your job for promotion, if you have multiple task groups in your canary
+deployment.
+
+-> **Note:** Using the ` + "`-prune=false`" + ` flag is recommended for this releaser. By default,
+Waypoint prunes and destroys all unreleased deployments and keeps only one previous
+deployment. Therefore, if ` + "`-prune=false`" + ` is not set, Waypoint may delete
+your job via "pruning" a previous version. See [deployment pruning](/docs/lifecycle/release#deployment-pruning)
+for more information.
+
+### Release URL
+
+If you want the URL of the release of your deployment to be published in Waypoint,
+you must set the meta 'waypoint.hashicorp.com/release_url' in your jobspec. The
+value specified in this meta field will be published as the release URL for your
+application. In the future, this may source from Consul.
+
+`)
+
+	doc.Example(`
+// The waypoint.hcl file
+release {
+	use "nomad-jobspec-canary" {
+		groups = [
+			"app"
+		]
+		fail_deployment = false
```

Comment:
I'm thinking since this is optional, we don't include it in the docs. If it's unset, it should default to false.